File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;

/**
 * 品牌接口
 * @author Administrator
 *
 */
public interface BrandService {

	public List<TbBrand> findAll();

	/**
	 * 分页查询
	 * @param pageNum 分页页数
	 * @param pageSize 每页行数
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);

	/**
	 * 新增商品种类
	 * @param tbBrand
	 */
	public  void addBrand(TbBrand tbBrand);


	/**
	 *
	 * @param tbBrand 修改后的参数
	 */
	public void updateBrand(TbBrand tbBrand);


	/**
	 * 根据id查询
	 * @param id
	 * @return
	 */
	public TbBrand findOne(long id);


	/**
	 * 多个删除
	 * @param ids
	 */
	public void deleteAll(Long[] ids);


    /**
     *
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
	public PageResult findPage(TbBrand tbBrand,int pageNum,int pageSize);


	/**
	 * 查找所有品牌的id和name
	 * @return
	 */
	List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }

    this.findPage=function (page,size) {
        return $http.get('../brand/findPage.do?pageNum='+page+'&pageSize='+size);
    }

    this.add=function(entity){
        return $http.post('../brand/add.do',entity);
    }

    this.update=function(entity){
        return $http.post('../brand/update.do',entity);
    }

    this.findOne=function(id){
        return $http.get('../brand/findOne.do?id='+id);
    }

    this.deleteAll=function (selectIds) {
        return $http.get('../brand/deletAll.do?ids='+selectIds);
    }

    this.search=function(page,size,searchEntity){
        return $http.post('../brand/search.do?pageNum='+page+'&pageSize='+size,searchEntity);
    }

    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }


})